### PR TITLE
[GEOT-7316] Child feature ClientProperties are replicated on GML parent container elements

### PR DESCRIPTION
**Pull request by fernandor777**
[GEOT-7316](https://osgeo-org.atlassian.net/browse/GEOT-7316) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Issue:
https://osgeo-org.atlassian.net/browse/GEOT-7316

The GML output comes with wrong repeated attributes inserted on container elements (gml:featureMember, wfs:member) pulled from the child Feature element.  This PR introduces a fix for this bug.
  
# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by aaime** [comment]
Legit QA test failure:

```
2023-03-07T04:19:03.9256796Z [INFO] PMD Failure: org.geotools.wfs.v2_0.WFSFeatureCollectionEncodingTest:48 Rule:wildcards Priority:3 No Wildcard Imports.
```

File: modules/extension/xsd/xsd-wfs/src/test/java/org/geotools/wfs/v2_0/WFSFeatureCollectionEncodingTest.java [hunk 1]
```diff
@@ -328,4 +329,26 @@ Encoder encoderFormatting() {
...
+        feature.add(0);
+        feature.featureUserData(Attributes.class, attributesMap);
+
+        WFSConfiguration configuration = new WFSConfiguration();
+        Encoder e = new Encoder(configuration);
+        e.getNamespaces().declarePrefix("geotools", "http://geotools.org");
+        e.setIndenting(true);
+
+        Document doc = e.encodeAsDOM(feature.buildFeature("id1"), WFS.member);
+        Assert.assertFalse(doc.getDocumentElement().hasAttribute("example"));
```
Comment (aaime) [inline modules/extension/xsd/xsd-wfs/src/test/java/org/geotools/wfs/v2_0/WFSFeatureCollectionEncodingTest.java:352]: Not checking that the feature itself has the attribute, like in the other test?

File: modules/extension/xsd/xsd-gml3/src/main/java/org/geotools/gml3/bindings/FeaturePropertyTypeBinding.java [hunk 2]
```diff
@@ -139,7 +139,6 @@ public Element encode(Object object, Document document, Element value) throws Ex
         if (object instanceof ComplexAttribute) {
             ComplexAttribute complex = (ComplexAttribute) object;
             checkXlinkHref(complex);
-            GML3EncodingUtils.encodeClientProperties(complex, value);
             GML3EncodingUtils.encodeSimpleContent(complex, document, value);
```
Comment (aaime) [inline modules/extension/xsd/xsd-gml3/src/main/java/org/geotools/gml3/bindings/FeaturePropertyTypeBinding.java:142]: Wondering about the simple content here... but I guess a feature would never have simple content inside... or would it?

**Review by aaime — CHANGES_REQUESTED** [review]
Seems ok, couple of questions, and check out the PMD failure above.

File: modules/extension/xsd/xsd-wfs/src/test/java/org/geotools/wfs/v2_0/WFSFeatureCollectionEncodingTest.java (hunk 1 above)
Comment (fernandor777) [reply modules/extension/xsd/xsd-wfs/src/test/java/org/geotools/wfs/v2_0/WFSFeatureCollectionEncodingTest.java:357]: > Legit QA test failure:
> 
> ```
> 2023-03-07T04:19:03.9256796Z [INFO] PMD Failure: org.geotools.wfs.v2_0.WFSFeatureCollectionEncodingTest:48 Rule:wildcards Priority:3 No Wildcard Imports.
> ```
Holy wildcards Batman!  Thanks fixed!

Added feature attribute check.

**Comment by aaime** [comment]
Thanks, merged!

**Pull request merged**